smartdashboard/src/edu/wpi/first/smartdashboard/gui/elements/Chooser.java

```diff
@@ -56,6 +56,10 @@ public void valueChanged(ITable source, String key, Object value, boolean isNew)
 		{
 			display.setSelected(source.getString(SELECTED));
 		}
+
+		if (source.getString(SELECTED) == null) {
+			source.putString(SELECTED, source.getString(DEFAULT, choices.get(0)));
+		}
```

Comment:
I think you need to use the getString(key, defaultValue) override. The robot by default does not set the key, so this would end up throwing an exception. `source.getString(SELECTED, null) == null` should work.